### PR TITLE
remove nans when setting sorted index

### PR DESCRIPTION
**Pull request by rockwellw**
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

**Comment by bnaul** [comment]
wow, so easy! the only edge case I can think of that wouldn't already be handled is if the whole thing is `NaN`, but presumably that already fails / probably should just fail anyway...?

File: dask/dataframe/tests/test_shuffle.py [hunk 1]
```diff
@@ -1078,8 +1078,4 @@ def test_shuffle_hlg_layer_serialize(npartitions):
 def test_set_index_nan_partition():
     d[d.a > 3].set_index("a")  # Set index with 1 null partition
     d[d.a > 1].set_index("a", sorted=True)  # Set sorted index with 0 null partitions
-    with pytest.raises(
-        ValueError,
-        match="Partitions with all-NaN index values are not supported for sorted indices.",
-    ):
-        d[d.a > 3].set_index("a", sorted=True)  # Set sorted index with 1 null partition
+    d[d.a > 3].set_index("a", sorted=True)  # Set sorted index with 1 null partition
```
Comment (mrocklin) [inline dask/dataframe/tests/test_shuffle.py:1081]: Can we try running the result through `assert_eq`, just to ensure that the result that this produces computes well?

```suggestion
    a = d[d.a > 3].set_index("a", sorted=True)  # Set sorted index with 1 null partition
    assert_eq(a, a)
```

**Comment by rockwellw** [comment]
> wow, so easy! the only edge case I can think of that wouldn't already be handled is if the whole thing is `NaN`, but presumably that already fails / probably should just fail anyway...?

Actually, that succeeds for me when testing locally, which seems like the correct behavior. You can set an index on an empty pandas df without issue. I'd argue that an empty df is inherently sorted, but I guess that's just semantics.

**Review by rjzamora — APPROVED** [review]
The change seems reasonable to me, and I can confirm that it works with cudf-base collections (dask_cudf) - Thanks!

**Pull request merged**